### PR TITLE
Provide a dockerfile to generate a container for the parsec backend

### PR DESCRIPTION
**Pull request by FirelightFlagboy**

### CODE REVIEW

**Review by touilleMan — COMMENTED** [review]
Concerning the name of the folder, I'd rather omit the `backend` given it's not a clear name.

So I think `packaging/server` is enough (given we also have `packaging/testbed-server`)

File: packaging/server/build-backend.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -xe
+
+# Remove bindings crates that aren't needed for backend-server
+sed -i -e '\;oxidation/bindings;d' Cargo.toml
+
+# Remove the line that generate pyqt code in the build script since the backend don't need it.
+sed -i -e '\;misc/generate_pyqt.py;d' build.py
```
Comment (touilleMan) [inline packaging/server/build-backend.sh:9]: This is too fragile, I'd rather introduce a `POETRY_PYQT_BUILD_STRATEGY=no_build` environ variable

File: packaging/server/build-backend.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# Install Rust (actual toolchain is going to be installed by maturin according to `rust-toolchain.toml`)
+curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain none
+
+export PATH="/root/.cargo/bin:$PATH"
+
+# Install parsec in virtual env
+python -m venv venv
+
+# Compile in release mode
+POETRY_LIBPARSEC_BUILD_PROFILE=release ./venv/bin/python -m pip install .[backend]
```
Comment (touilleMan) [inline packaging/server/build-backend.sh:20]: `POETRY_LIBPARSEC_BUILD_PROFILE` default value is already `release`, but I guess better be extra careful on this ! ^^

File: packaging/server/build-backend.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```
Comment (touilleMan) [inline packaging/server/build-backend.sh:1]: I'd rather move this script in a subfolder (`bin` ? `res` ? `resources` ?)
The idea is to make it obvious this is a resource for the dockerfile and not a script that should be directly run by the dev

File: packaging/server/build.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+SCRIPTDIR=$(dirname $(realpath -s "$0"))
+ROOTDIR="$SCRIPTDIR/../.."
+
+CURRENT_DATE=$(date --iso-8601)
+CURRENT_VERSION=$(grep -o '^version = .*$' $ROOTDIR/pyproject.toml | sed 's/version = "\(.*\)"$/\1/' | tr '+' '.')
+CURRENT_COMMIT_SHA=$(git rev-parse --short HEAD)
+
+UNIQ_TAG="$CURRENT_DATE-$CURRENT_VERSION-$CURRENT_COMMIT_SHA"
+PREFIX=ghcr.io/scille/parsec-cloud
```
Comment (touilleMan) [inline packaging/server/build.sh:14]: Would be good to add a small comment about this prefix, from what I understand `ghcr.io/scille/parsec-cloud` refers to Github's own docker images repository for our project

File: packaging/server/build.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+echo
+
+DOCKER_BUILDKIT=1 docker build \
+    -f $SCRIPTDIR/server.dockerfile \
+    -t $PREFIX/parsec-backend-server:latest \
+    -t $PREFIX/parsec-backend-server:$UNIQ_TAG \
+    $ROOTDIR $@
+
+echo
+echo "You can now test/use the docker image with:"
```
Comment (touilleMan) [inline packaging/server/build.sh:30]: Very nice help 👍

File: packaging/server/server.dockerfile.dockerignore [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+htmlcov
+
+# Other files
+docs
+json_schema
+logs
+licenses
+newsfragment
+packaging
+!packaging/server/build-backend.sh
```
Comment (touilleMan) [inline packaging/server/server.dockerfile.dockerignore:34]: `!packaging/server` ? would avoid silly bug next time we add (or rename) files

**Review by touilleMan — CHANGES_REQUESTED** [review]

File: packaging/server/build.sh (hunk 4 above)
Comment (FirelightFlagboy) [reply packaging/server/build.sh:14]:
```suggestion
# We use Github package repository to store our docker's container.
PREFIX=ghcr.io/scille/parsec-cloud
```

WDYT ?

File: packaging/server/server.dockerfile.dockerignore (hunk 6 above)
Comment (FirelightFlagboy) [reply packaging/server/server.dockerfile.dockerignore:34]: Done

File: packaging/server/build-backend.sh (hunk 3 above)
Comment (FirelightFlagboy) [reply packaging/server/build-backend.sh:1]: done

File: packaging/server/build.sh (hunk 4 above)
Comment (touilleMan) [reply packaging/server/build.sh:14]: 👍

File: packaging/server/build-backend.sh (hunk 1 above)
Comment (FirelightFlagboy) [reply packaging/server/build-backend.sh]: done

**Review by touilleMan — APPROVED** [review]

**Pull request merged**